File: guests/go/grow/grow.go
// The grow conformance scene (tools/scenes/grow.steps). EVERY CHILD IS A
// GROWER, or the split is no longer weight/Σweight.
package grow

import (
	kaya "dev.kaya/bindings/go"
)

func App() *kaya.App {
	app := kaya.NewApp()

	app.Build(func(tx *kaya.Tx) {
		probe := tx.Signal("grow probe")
		one := tx.Signal("one")

		tx.Mount(tx.Column(func() {
			tx.Label(probe).Grow(1)   // label#0
			tx.Textarea(nil).Grow(2)  // textarea#0
			tx.Row(func() {
				tx.Label(one).Grow(1) // label#1
				tx.Button("three", nil).Grow(3)
			}).Grow(1).Spacing(12)
		}))
	})

	return app
}

File: guests/java/dev/kaya/guests/Grow.java
package dev.kaya.guests;

import dev.kaya.KayaApp;

/**
 * The grow conformance scene from the JVM (KAYA_SELFTEST=grow) — see
 * guests/rust/grow.rs for the rationale. Every child is a grower, so
 * each split is exactly weight/Σweight: 1,2,1 divide the column
 * 25/50/25 and the row's 1,3 divide its width 25/75.
 */
public final class Grow {
    public static void app() {
        KayaApp app = new KayaApp();

        app.build(tx -> {
            KayaApp.Signal<String> probe = tx.signal("grow probe");
            KayaApp.Signal<String> one = tx.signal("one");

            tx.mount(tx.column(() -> {
                tx.label(probe).grow(1.0); // label#0
                tx.textarea().grow(2.0); // textarea#0
                tx.row(() -> {
                    tx.label(one).grow(1.0); // label#1
                    tx.button("three").grow(3.0);
                }).grow(1.0).spacing(12.0);
            }));
            return null;
        });

        app.dispatchLoop();
    }

    private Grow() {}
}
